tests/org.eclipse.passage.lic.net.tests/src/org/eclipse/passage/lic/net/tests/io/TestKeyKeeperFactory.java

```diff
@@ -0,0 +1,20 @@
+package org.eclipse.passage.lic.net.tests.io;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+import org.eclipse.passage.lic.api.LicensingException;
+import org.eclipse.passage.lic.api.io.KeyKeeper;
+import org.eclipse.passage.lic.base.io.FileKeyKeeper;
+
+class TestKeyKeeperFactory {
```

Comment:
_Either abstract or final_ means this class must be `abstract`.
Constructor is `protected`. 
Any not-empty implementation - again `final`. 

In this case there must be an interface with two implementations (or abstract parent class in is there are something useful to share between implementations).

Reply:
The subclass uses the `publicKey()` method, therefore I chose an abstract super-class.